### PR TITLE
ref: engineio base

### PR DESCRIPTION
**Pull request by mrfoe7**

### CODE REVIEW

File: engineio/base/conn.go [hunk 1]
```diff
@@ -99,10 +31,10 @@ type ConnParameters struct {
 }
 
 type jsonParameters struct {
-	SID          string   `json:"sid"`
-	Upgrades     []string `json:"upgrades"`
 	PingInterval int      `json:"pingInterval"`
 	PingTimeout  int      `json:"pingTimeout"`
+	SID          string   `json:"sid"`
```
Comment (mrfoe7) [inline engineio/base/conn.go:36]: notice: memory align struct

File: engineio/base/conn_test.go [hunk 2]
```diff
@@ -37,45 +38,52 @@ func TestOpError(t *testing.T) {
 		temporary bool
 		errString string
 	}{
-		{"http://domain/abc", "post(write) to", io.EOF, false, false, "post(write) to http://domain/abc: EOF"},
-		{"http://domain/abc", "get(read) from", io.EOF, false, false, "get(read) from http://domain/abc: EOF"},
-		{"http://domain/abc", "post(write) to", fakeOpError{true, false}, true, false, "post(write) to http://domain/abc: fake error"},
-		{"http://domain/abc", "get(read) from", fakeOpError{false, true}, false, true, "get(read) from http://domain/abc: fake error"},
+		{"http://domain/abc", "post(write) to", io.EOF, false, false,
```
Comment (googollee) [inline engineio/base/conn_test.go:41]: If wrapping test cases, please cut them in the same column. I prefer to cut at `timeout` field.

File: engineio/base/packet_test.go [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+var testPacketTypeCases = []struct {
```
Comment (googollee) [inline engineio/base/packet_test.go:11]: As other tests don't use these cases, I prefer to put it in the `testPacketTypeCases` function.

File: engineio/base/conn_test.go (hunk 2 above)
Comment (mrfoe7) [reply engineio/base/conn_test.go:41]: Sorry, i don't understand. What are you suggest?

**Review by googollee — APPROVED** [review]

File: engineio/base/conn_test.go (hunk 2 above)
Comment (googollee) [reply engineio/base/conn_test.go:41]:
```
{"http://domain/abc", "post(write) to", io.EOF,
	false, false, "post(write) to http://domain/abc: EOF"},
{"http://domain/abc", "get(read) from", io.EOF,
	 false, false, "get(read) from http://domain/abc: EOF",
},
{"http://domain/abc", "post(write) to", fakeOpError{true, false},
	true, false, "post(write) to http://domain/abc: fake error",
},
{"http://domain/abc", "get(read) from", fakeOpError{false, true},
	false, true, "get(read) from http://domain/abc: fake error",
}
```

Something like that. All cases wrap at `timeout` field.

Comment (mrfoe7) [reply engineio/base/conn_test.go]: fixed

**Review by googollee — APPROVED** [review]

**Comment by erkie** [comment]
Nice cleanup!

**Pull request merged**